app/src/main/java/Fragments/ProfileFragment.java

```diff
@@ -22,17 +48,105 @@ public static ProfileFragment newInstance() {
     }
 
     @Override
-    public void onCreate(Bundle savedInstanceState)
-    {
+    public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
     }
 
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_profile, container, false);
+
+        name_tv = (TextView) view.findViewById(R.id.profile_name);
+        email_tv = (TextView) view.findViewById(R.id.profile_email);
+        phoneNumber_tv = (TextView) view.findViewById(R.id.profile_phone_number);
+        photo_iv = (ImageView) view.findViewById(R.id.profile_photo);
+        addPhoto_ib = (ImageButton) view.findViewById(R.id.profile_add_photo);
+
+        if (user != null) {
+            name = user.getDisplayName();
+            email = user.getEmail();
+//            photoUrl = user.getPhotoUrl();
+            // tmp
+            photoUrl = Uri.parse("http://i0.kym-cdn.com/photos/images/original/000/692/145/49c.png");
+
+            // TODO: Check if user's email is verified?
+
+            if (name != null) {
+                name_tv.setText(name);
+            }
+            else {
+                name_tv.setText("Evan Mateo");
+//                editName();
+//                name = user.getDisplayName();
+//                name_tv.setText(name);
+            }
+
+            email_tv.setText(email);
+
+            new DownloadImageTask(photo_iv).execute(photoUrl.toString());
+
+            name_tv.setVisibility(View.VISIBLE);
+            email_tv.setVisibility(View.VISIBLE);
+            phoneNumber_tv.setVisibility(View.VISIBLE);
+            photo_iv.setVisibility(View.VISIBLE);
+            addPhoto_ib.setVisibility(View.VISIBLE);
+        } else {
+            name_tv.setVisibility(View.GONE);
+            email_tv.setVisibility(View.GONE);
+            phoneNumber_tv.setVisibility(View.GONE);
+            photo_iv.setVisibility(View.GONE);
+            addPhoto_ib.setVisibility(View.GONE);
+
+            // TODO: Display message telling user that they are currently not logged in. Suggest signing up or logging in.
+        }
+
         // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_profile, container, false);
+        return view;
     }
 
+    @Override
+    public void onStart() {
+        super.onStart();
+
+        // TODO: Set listener for add_photo ImageButton to call addPhoto()
+        addPhoto_ib.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                editPhoto();
+            }
+        });
+
+        // TODO: Set listener for edit_email, edit_phone_number
+    }
 
```

Comment:
Maybe this section just add comments such as User Profile Edit, but code is clean so easy to read!